File: basic-algorithms/js/08-repeat-string.js
function repeatStringNumTimes(str, num, bnoden) {
	bnoden = ''
	if (!isNaN(num) && num > 0) bnoden = str.repeat(num)
	return bnoden
}

console.log(repeatStringNumTimes("*", 3)) // ***
console.log(repeatStringNumTimes("abc", 3)) // abcabcabc
console.log(repeatStringNumTimes("abc", 4)) // abcabcabcabc
console.log(repeatStringNumTimes("abc", 1)) // abc
console.log(repeatStringNumTimes("*", 8)) // ********
console.log(repeatStringNumTimes("abc", -2)) // ""

File: basic-algorithms/08repeatstring.go
// https://play.golang.org/p/3XWkf0h_Xw
package main

import (
	"fmt"
	"strings"
)

func repeatStringNumTimes(str string, num int) string {
	bnoden := ""
	if num > 0 {
		bnoden = strings.Repeat(str, num)
	}
	return bnoden
}

// Test
func main() {
	fmt.Println(repeatStringNumTimes("*", 3))    // ***
	fmt.Println(repeatStringNumTimes("abc", 3))  // abcabcabc
	fmt.Println(repeatStringNumTimes("abc", 4))  // abcabcabcabc
	fmt.Println(repeatStringNumTimes("abc", 1))  // abc
	fmt.Println(repeatStringNumTimes("*", 8))    // ********
	fmt.Println(repeatStringNumTimes("abc", -2)) // ""
}
